### PR TITLE
[IA-1637] Hail jobs fail with "mismatched python version"

### PR DESCRIPTION
**Pull request by rtitle**
Corresponding terra-docker PR: https://github.com/DataBiosphere/terra-docker/pull/99

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/resources/bucket-tests/hail_samples.csv [hunk 1]
```diff
@@ -0,0 +1,2505 @@
+,sample,age,ancestry,bmi,fg,fi,hdl,height,ldl,population,sex,tc,tg,whr
```
Comment (rtitle) [inline automation/src/test/resources/bucket-tests/hail_samples.csv:1]: This is synthetic data, and the same as in this workspace: https://app.terra.bio/#workspaces/fc-product-demo/2019_ASHG_Reproducible_GWAS

For some reason I couldn't reproduce the Hail error on a small (5x5) dataframe. But this data does reproduce it. Maybe the dataset needs to be larger in order to be distributed to the workers -- just guessing.

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala [hunk 2]
```diff
@@ -116,5 +116,43 @@ class NotebookHailSpec extends ClusterFixtureSpec with NotebookTestUtils {
...
+                  |hl.init()
+                """.stripMargin
+              notebookPage.executeCell(importHail)
+
+              // Import the DataFrame into a Hail table
+              val result = notebookPage.executeCell(s"samples = hl.Table.from_pandas(df, key = 'sample')")
+              result shouldBe 'defined
+              result.get should not include ("FatalError")
+              result.get should not include ("PythonException")
+              // TODO what's a positive verification?
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala]: I'll add some more assertions here

File: jenkins/Jenkinsfile [hunk 3]
```diff
@@ -279,9 +279,13 @@ pipeline {
           script {
             for(String dp_version: dataproc_versions.trim().split(' ')) {
               def dp_version_formatted = dp_version.replace(".","-")
+              // Dataproc 1.2 is used for the legacy image and must have python 3.6 installed
```
Comment (rtitle) [inline jenkins/Jenkinsfile:282]: This is a little hacky, but couldn't think of a cleaner way. Basically we want:
1. legacy image -- dataproc 1.2 -- python 3.6.8 on the workers
2. terra-docker image -- dataproc 1.4 -- python 3.7.4 on the workers

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala (hunk 2 above)
Comment (kyuksel) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala]: Is there a way to get a slice of the frame to use as expected value in positive verification?

Comment (kyuksel) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala]: It looks like you're doing that below already so it should be good?

**Review by kyuksel — COMMENTED** [review]
Did you get to verify the Jenkins job with both the legacy and the non-legacy images to make sure they result in the right version of Python installed on workers?

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala (hunk 2 above)
Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookHailSpec.scala]: Ah, the below assertion was copy-pasted and probably not correct (it should have more than 4 rows). This is WIP, still playing around with it.

**Comment by rtitle** [comment]
> Did you get to verify the Jenkins job with both the legacy and the non-legacy images to make sure they result in the right version of Python installed on workers?

Yup running here: https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-dataproc-image/268/console, when it's done I'll try out the generated images.

**Review by Qi77Qi — APPROVED** [review]

File: jenkins/Jenkinsfile [hunk 4]
```diff
@@ -173,7 +173,7 @@ pipeline {
 
     stage('Leonardo Git') {
       steps {
-        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/leonardo.git', branch: 'develop'
+        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/leonardo.git', branch: 'rt-ia-1637'
```
Comment (rtitle) [inline jenkins/Jenkinsfile]: TODO revert me!

**Comment by rtitle** [comment]
fiab-start failed, jenkins retest

**Pull request merged**